crates/cli/src/cli.rs

```diff
@@ -95,6 +96,50 @@ fn main() -> anyhow::Result<()> {
         }
     };
 
+    // -- Network selection --
+
+    // Custom network selection
+    let custom_internal_api_endpoint_url = env::var("CUSTOM_INTERNAL_API_ENDPOINT_URL");
```

Comment:
Hey this is good, however, since we separate the user's interaction in sections, please move this after line 154, right before the class hash check.

Reply:
@cwkang1998 Can you review it again?